src/aggregator/aggregator/elem_base.go

```diff
@@ -230,63 +237,114 @@ type elemMetrics struct {
 	scope         tally.Scope
 	updatedValues tally.Counter
 	retriedValues tally.Counter
-	forwardLags   map[forwardLagKey]tally.Histogram
+	flush         map[flushKey]flushMetrics
 	mtx           sync.RWMutex
 }
 
-type forwardLagKey struct {
-	resolution    time.Duration
-	listType      metricListType
-	flushType     flushType
-	fwdType       string
-	jitterApplied bool
+// flushMetrics are the metrics produced by a flush task processing the metric element.
+type flushMetrics struct {
+	// count of element scanned.
+	elemsScanned tally.Counter
+	// count of values (i.e aggregated timestamps) processed.
+	valuesProcessed tally.Counter
+	// count of values expired.
+	valuesExpired tally.Counter
+	// the difference between actual and expected processing for a value.
+	forwardLags map[forwardKey]tally.Histogram
+}
+
+func newFlushMetrics(scope tally.Scope) flushMetrics {
+	forwardLagBuckets := tally.DurationBuckets{
+		10 * time.Millisecond,
+		500 * time.Millisecond,
+		time.Second,
+		2 * time.Second,
+		5 * time.Second,
+		10 * time.Second,
+		15 * time.Second,
+		20 * time.Second,
+		25 * time.Second,
+		30 * time.Second,
+		35 * time.Second,
+		40 * time.Second,
+		45 * time.Second,
+		60 * time.Second,
+		90 * time.Second,
+		120 * time.Second,
```

Comment:
Might be useful to have this go beyond 5m if we expect resolutions above that

Reply:
true, since it includes jitter.

Reply:
this makes me realize that the lag time _with_ jitter is pretty useless for high resolution (5m). we won't have enough buckets to understand the actual lag.